shopfloor/services/cluster_picking.py

```diff
@@ -992,13 +994,16 @@ def change_pack_lot(self, picking_batch_id, move_line_id, barcode):
                 move_line, lot, response_ok_func, response_error_func
             )
             if response:
+                response["data"]["scan_destination"].update({"qty_done": quantity})
                 return response
 
         package = search.package_from_scan(barcode)
         if package:
-            return change_package_lot.change_package(
+            response = change_package_lot.change_package(
                 move_line, package, response_ok_func, response_error_func
             )
+            response["data"]["scan_destination"].update({"qty_done": quantity})
+            return response
```

Comment:
Not sure it is correct, you expect the `response` to get the `scan_destination` state to set the `qty_done`, but the response could be returned by either `response_ok_func` or `response_error_func` here (respectively `self._response_for_scan_destination` and `self._response_for_change_pack_lot`).
Better to test first if `scan_destination` is present.